src/pcre2_jit_compile.c

```diff
@@ -13107,9 +13104,25 @@ ccend = bracketend(common->start);
 
 /* Calculate the local space size on the stack. */
 common->ovector_start = LOCAL0;
-common->optimized_cbracket = (sljit_u8 *)SLJIT_MALLOC(re->top_bracket + 1, allocator_data);
-if (!common->optimized_cbracket)
+/* Allocate space for temporary data structures. */
+private_data_length = ccend - common->start;
+/* The chance of overflow is very low, but might happen on 32 bit. */
+if (private_data_length > ~(sljit_uw)0 / sizeof(sljit_s32))
+  return PCRE2_ERROR_NOMEMORY;
+
+private_data_length *= sizeof(sljit_s32);
+/* Align to 32 bit. */
+total_length = ((re->top_bracket + 1) + (sljit_uw)(sizeof(sljit_s32) - 1)) & ~(sljit_uw)(sizeof(sljit_s32) - 1);
+if (~(sljit_uw)0 - private_data_length <= total_length)
```

Comment:
nit: I think this could be `<` rather than `<=`

Reply:
Since max is odd number, and the other two are even, the == cannot happen. Ok, lets change it.